src/openrct2/ride/Vehicle.cpp

```diff
@@ -8859,21 +8856,19 @@ void Vehicle::UpdateTrackMotionMiniGolfVehicle(Ride* curRide, rct_ride_entry* ri
     if (!IsHead())
     {
         Vehicle* prevVehicle = GET_VEHICLE(prev_vehicle_on_ride);
-        regs.al = prevVehicle->TrackSubposition;
-        if (regs.al != VEHICLE_TRACK_SUBPOSITION_MINI_GOLF_START_9)
+        TrackSubposition = prevVehicle->TrackSubposition;
+        if (TrackSubposition != VEHICLE_TRACK_SUBPOSITION_MINI_GOLF_START_9)
         {
-            regs.al--;
+            TrackSubposition--;
         }
-        TrackSubposition = regs.al;
     }
 
     ClearUpdateFlag(VEHICLE_UPDATE_FLAG_ON_LIFT_HILL);
     track_type = (tileElement->AsTrack()->GetTrackType() << 2) | (direction & 3);
     var_CF = tileElement->AsTrack()->GetBrakeBoosterSpeed();
-    regs.ax = 0;
+    track_progress = 0;
```

Comment:
Are you *entirely* sure that the contents of regs.ax are *always* written to `track_progress`?

Reply:
Yes.

What happened here is that on label loc_6DC743 we had track_progress = regs.ax
What I did was find all places that went to that label (only two places) and moved the assignment there. Then I cleaned up the register use.

Reply:
Same changes were made on the non golf version. Looks fine to me